### PR TITLE
Brought Up To Standard

### PR DESCRIPTION
**Pull request by XedinUnknown**
Aims to incorporate solution for #8.

### CODE REVIEW

**Review by mecha — CHANGES_REQUESTED** [review]
The second requested change in #8 is still not fulfilled in this PR.

**Comment by XedinUnknown** [comment]
Sorry, forgot to mention that it's WIP. Early review, so to speak.

File: src/SpecAwareTrait.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * @since [*next-version*]
+ */
+trait SpecAwareTrait
+{
+    /**
+     * The spec.
+     *
+     * @since [*next-version*]
+     *
+     * @var mixed
```
Comment (mecha) [inline src/SpecAwareTrait.php]: Typehint as `array|Traversable|null`

File: src/SpecAwareTrait.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    protected $spec;
+
+    /**
+     * Retrieves the spec associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return array|Traversable The spec.
```
Comment (mecha) [inline src/SpecAwareTrait.php]: Add `null` as a possible return type.

**Review by mecha — APPROVED** [review]

**Comment by mecha** [comment]
The doc tag order is off (such as in `AbsractCompositeValidator`). I suggest running the CS fixer; afaik it should clean these up.

**Review by mecha — APPROVED** [review]
LGTM

**Comment by XedinUnknown** [comment]
@mecha, CS fixer doesn't seem to care whether param docs come before or after the `@since`. If it can't be done automatically, I wouldn't bother.

**Pull request merged**